go/src/lib.rs

```diff
@@ -306,6 +304,70 @@ fn convert_vec_to_pointer<T>(mut vec: Vec<T>) -> (*mut T, c_long) {
     (vec_ptr, len)
 }
 
+fn valkey_value_to_command_response(value: Value) -> RedisResult<Option<CommandResponse>> {
+    let mut command_response = CommandResponse::default();
+    let result: RedisResult<Option<CommandResponse>> = match value {
+        Value::Nil => Ok(None),
+        Value::SimpleString(text) => {
+            let vec = text.chars().map(|b| b as c_char).collect::<Vec<_>>();
+            let (vec_ptr, len) = convert_vec_to_pointer(vec);
+            command_response.string_value = vec_ptr;
+            command_response.string_value_len = len;
+            Ok(Some(command_response))
+        }
+        Value::BulkString(text) => {
+            let vec = text.iter().map(|b| *b as c_char).collect::<Vec<_>>();
+            let (vec_ptr, len) = convert_vec_to_pointer(vec);
+            command_response.string_value = vec_ptr;
+            command_response.string_value_len = len;
+            Ok(Some(command_response))
+        }
+        Value::VerbatimString { format: _, text } => {
+            let vec = text.chars().map(|b| b as c_char).collect::<Vec<_>>();
+            let (vec_ptr, len) = convert_vec_to_pointer(vec);
+            command_response.string_value = vec_ptr;
+            command_response.string_value_len = len;
+            Ok(Some(command_response))
+        }
+        Value::Okay => {
+            let vec = "OK".chars().map(|b| b as c_char).collect::<Vec<_>>();
+            let (vec_ptr, len) = convert_vec_to_pointer(vec);
+            command_response.string_value = vec_ptr;
+            command_response.string_value_len = len;
+            Ok(Some(command_response))
+        }
+        Value::Int(num) => {
+            command_response.int_value = num;
+            Ok(Some(command_response))
+        }
+        Value::Double(num) => {
+            command_response.float_value = num;
+            Ok(Some(command_response))
+        }
+        Value::Boolean(boolean) => {
+            command_response.bool_value = boolean;
+            Ok(Some(command_response))
+        }
+        Value::Array(array) => {
+            let vec: Vec<CommandResponse> = array
+                .into_iter()
+                .map(|v| {
+                    valkey_value_to_command_response(v)
```

Comment:
General comment about Rust code here:

This will crash the client. We should avoid `expect()` and `unwrap()`. 
We should go with what we did in Java where we returned `Result<..>` instead of raw value and handle it in a common place. In Java we return a Java exception, I am not sure what we should return. What I do know is that we must not call `unwrap` / `expect`

Reply:
Noted, I have added a TODO for now to address this in the upcoming PR as the findings could take time and would block upcoming PRs as well.